README.md

```diff
@@ -1,12 +1,189 @@
+    case Action.ERROR:
+      return { ...state, error: action.error, isLoading: false };
+    default:
+      return state;
+  }
+};
+
+const useQuery = <T>(query: string, options: RequestInit = {}): UseQueryState<T> => {
+  const initialState: UseQueryState<T> = {
+    data: null,
+    error: null,
+    isLoading: true,
+  };
+  const reducer = makeReducer<T>();
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    // loading
+    dispatch({ type: Action.LOADING });
+    try {
+      const res = await fetch('https://pokeapi-graphiql.herokuapp.com/', {
+        ...options,
+        body: JSON.stringify({ query }),
+        headers: {
+          ...(options.headers || {}),
+          'Content-Type': 'application/json',
+        },
+        method: 'POST',
+      });
+      const { data }: { data: T } = await res.json(); // Note: some servers may return errors here as well
+      // success
+      dispatch({ data, type: Action.SUCCESS });
+    } catch (error) {
+      // error
+      dispatch({ error: error.message, type: Action.ERROR });
+    }
+  });
+};
+
+const ListPokemon: React.FunctionComponent = () => {
+  const { isLoading, data, error } = useQuery<Pokemon[]>(gql`
+    query POKEMON {
+      pokemons(first: 151) {
+        number
+        name
+      }
+    }
+  `);
+
+  if (isLoading) return <div>Loading…</div>;
+  if (error) return <div>{error}</div>;
+
+  return (
+    <table>
+      <thead>
+        <tr>
+          <th>Pokémon Name</th>
+          <th>Number</th>
+        </tr>
+      </thead>
+      <tbody>
+        {data.map(({ name, number }) => (
+          <tr key={name}>
+            <td>{name}</td>
+            <td>{number}</td>
+          </tr>
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+export default ListPokemon;
+```
+
+## Comparison
+
+Is comparing gql-zero fair to compare to other libraries? **No! This is just
+a simple string template!** It doesn’t come close to the features in other
+libraries. In that regard, this library is a slimmer version of
+[graphql-tag][graphql-tag]
+
+Still, this chart is a handy reference because many people often don’t
+consider the full weight of their GraphQL solution on the client based on
+deps and peerDeps.
+
+If you don’t need the extra features, don’t force your users to download
+them!
+
+| Library                |   Total weight | Avoids `graphql` |
+| :--------------------- | -------------: | :--------------: |
+| `@manifoldco/gql-zero` | 🔥 `0.4 KB` 🔥 |        ✅        |
+| `@apollo/react-hooks`  |      `~ 60 KB` |        ✅        |
+| `apollo-boost` \*      |     `263.6 KB` |        🚫        |
+| `apollo-client`        |      `56.9 KB` |        ✅        |
+| `apollo-react`         |      `71.6 KB` |        ✅        |
+| `graphql-tag`          |       `1.9 KB` |        ✅        |
+| `micro-graphql-react`  |       `8.8 KB` |        ✅        |
+| `urql`                 |      `21.5 KB` |        ✅        |
```

Comment:
I'm confused by this table. I thought most of these at least had `graphql` as a peer dependency. In particular, I could not use `graphql-tag` without downloading `graphql`, or I'd get an error.

Reply:
Ohh you’re right—I missed that `graphql` is a peerDep of `graphql-tag` 🤔